src/solvers/smt2/smt2_conv.cpp

```diff
@@ -2890,23 +2896,15 @@ void smt2_convt::convert_constant(const constant_exprt &expr)
     UNEXPECTEDCASE("unknown constant: "+expr_type.id_string());
 }
 
-void smt2_convt::convert_mod(const mod_exprt &expr)
+SMT_Term smt2_convt::convert_mod(const mod_exprt &expr)
 {
-  if(expr.type().id()==ID_unsignedbv ||
-     expr.type().id()==ID_signedbv)
-  {
-    if(expr.type().id()==ID_unsignedbv)
-      out << "(bvurem ";
-    else
-      out << "(bvsrem ";
+  PRECONDITION(
+    expr.type().id() == ID_unsignedbv || expr.type().id() == ID_signedbv);
 
-    convert_expr(expr.op0());
-    out << " ";
-    convert_expr(expr.op1());
-    out << ")";
-  }
+  if(expr.type().id() == ID_unsignedbv)
```

Comment:
The fact that this has operands that need to be `convert_expr`'d means it's not a leaf right?